homework-submission/main.js

```diff
@@ -0,0 +1,67 @@
+const List = props => {
+  return (
+    <li>
+      {props.description},{props.deadline}
+    </li>
+  );
+};
+
+class TodoList extends React.Component {
+  staticList = [
+    { description: 'Get out of bed', deadline: 'Wed Sep 13 2017' },
+    { description: 'Brush teeth', deadline: 'Thu Sep 14 2017' },
+    { description: 'Eat breakfast', deadline: 'Fri Sep 15 2017' },
+  ];
+
+  dynamicList = [
+    {
+      id: 1,
+      description: 'Get out of bed',
+      deadline: '2017-09-11',
+      done: true,
+    },
+    {
+      id: 2,
+      description: 'Brush teeth',
+      deadline: '2017-09-10',
+      done: false,
+    },
+    {
+      id: 3,
+      description: 'Eat breakfast',
+      deadline: '2017-09-09',
+      done: false,
+    },
+  ];
+
+  render() {
+    return (
+      <div>
+        <h1>static List</h1>
+        <List description={this.staticList[0].description} deadline={this.staticList[0].deadline} />
+        <List description={this.staticList[1].description} deadline={this.staticList[1].deadline} />
+        <List description={this.staticList[2].description} deadline={this.staticList[2].deadline} />
+
+        <div>
+          <h1>Dynamic List</h1>
+          {this.dynamicList.map(todoList => (
+            <div>
+              <li
+                key={todoList.id}
```

Comment:
Having the key anywhere isn't the proper use of it, having it in the very first element of the mapped array _(line 48)_ will remove the following error:

![Screen Shot 2019-07-04 at 03 27 30](https://user-images.githubusercontent.com/25272822/60633590-af09e180-9e0b-11e9-8a60-651e1d75512a.png)

It's very important to note, Having a key is the React way of identifying each element in the array of it's fellow, And since `.map` method returning an array of elements then it's a valid syntax to say:
```javascript
[
  <p>hello world</p>
]
```

and as a correction for the key thing you may look at the following:
```javascript
// Bad
[
  <div><p key="some unique key">hello world</p></div>
]

// Good
[
  <div key="some unique key"><p>hello world</p></div>
]
```